react-components/intro-to-react-js.md

```diff
@@ -0,0 +1,101 @@
+- Display and handle the UI to create a new post
+
+![](../assets/react-components_intro-to-react-js_reddit.png)
+
+Imagine creating all of this functionality using HTML and vanilla JavaScript. It wouldn't be impossible, but our resulting code is probably:
+
+- Huge! There's a lot of code in that logic to write!
+- Unorganized. Is there one file that holds all the JS? Are there multiple files? How are they organized?
+- Full of repetition. We need to create event handlers all the time. Is there a way to simplify this repetition?
+
+Great software is maintainable, functional, and performant. How can we strive towards those values in our front-end code?
+
+## Vocabulary and Synonyms
+
+| Vocab               | Definition                                                                                                                                                                                               | How to Use in a Sentence                                                                                                                                                                                               |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| User Interface (UI) | The elements that a user uses in order to interact with a machine. UI elements are usually visual, but UI also describes _how_ a user interacts with a webapp, such as buttons, text fields, or sliders. | "This UI was easy to use and enjoyable, because the text was readable, and the buttons were intuitive," "This UI was difficult to use, because I didn't understand the navigation menu, and the data was unorganized." |
+| Components (React)  | Independent, reusable pieces of UI.                                                                                                                                                                      | "My recipe webapp has a `Recipe` component, which renders a recipe," "I created a `RedButton` component to reuse. All `RedButton` components are red and animate on click."                                            |
+
+## Create Complex UIs with React Components
+
+React is an [open-source](https://github.com/facebook/react) JavaScript library for building user interfaces.
+
+The React library will give users the ability to create React _components_.
+
+**Components** are rendered UI elements. Components in React may contain:
+
+- Styles
+- Complex logic
+- Data
+- State
+- Event-handling
+
+Components are intended to be flexible and reusable, too! A single web page can render many components inside it.
+
```

Comment:
suggest a callout to stress that component is an overloaded term, and _this_ is the meaning we are using for now.
```suggestion

### !callout-info

## Component Is an Over-Used Programming Term

We run into the term _component_ a lot when programming. It doesn't always mean exactly the same thing in every situation. When we are discussing React, we should focus on components as being independent, reusable pieces of UI, regardless of what component might mean in other contexts.

### !end-callout

```

Reply:
THANK YOU!